examples/fluids/src/misc.c

```diff
@@ -465,3 +466,175 @@ PetscErrorCode IntArrayP2C(PetscInt num_entries, PetscInt **array_petsc, CeedInt
+  }
+  // Mesh
+  const PetscInt num_comp_q = 5;
+  PetscInt       glob_dofs, owned_dofs, local_dofs;
+  const CeedInt  num_P = user->app_ctx->degree + 1, num_Q = num_P + user->app_ctx->q_extra;
+  // -- Get global size
+  PetscCall(DMGetGlobalVectorInfo(user->dm, &owned_dofs, &glob_dofs, NULL));
+  // -- Get local size
+  PetscCall(DMGetLocalVectorInfo(user->dm, &local_dofs, NULL, NULL));
+  PetscCall(PetscPrintf(comm,
+                        "  Mesh:\n"
+                        "    Number of 1D Basis Nodes (P)       : %" CeedInt_FMT "\n"
+                        "    Number of 1D Quadrature Points (Q) : %" CeedInt_FMT "\n"
+                        "    Global DoFs                        : %" PetscInt_FMT "\n"
+                        "    DoFs per node                      : %" PetscInt_FMT "\n"
+                        "    Global %" PetscInt_FMT "-DoF nodes                 : %" PetscInt_FMT "\n",
+                        num_P, num_Q, glob_dofs, num_comp_q, num_comp_q, glob_dofs / num_comp_q));
+  // -- Get Partition Statistics
+  PetscCall(PetscPrintf(comm, "  Partition:                             (min,max,median,max/median)\n"));
+  {
+    PetscInt *gather_buffer = NULL;
+    PetscInt  part_owned_dofs[3], part_local_dofs[3], part_boundary_dofs[3], part_neighbors[3];
+    PetscInt  median_index = comm_size % 2 ? comm_size / 2 : comm_size / 2 - 1;
+    if (!rank) PetscCall(PetscMalloc1(comm_size, &gather_buffer));
+
+    PetscCallMPI(MPI_Gather(&owned_dofs, 1, MPIU_INT, gather_buffer, 1, MPIU_INT, 0, comm));
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
+      part_owned_dofs[0]             = gather_buffer[0];              // min
+      part_owned_dofs[1]             = gather_buffer[comm_size - 1];  // max
+      part_owned_dofs[2]             = gather_buffer[median_index];   // median
+      PetscReal part_owned_dof_ratio = (PetscReal)part_owned_dofs[1] / (PetscReal)part_owned_dofs[2];
+      PetscCall(PetscPrintf(
+          comm, "    Owned free %" PetscInt_FMT "-DoF nodes             : %" PetscInt_FMT ", %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n", num_comp_q,
+          part_owned_dofs[0] / num_comp_q, part_owned_dofs[1] / num_comp_q, part_owned_dofs[2] / num_comp_q, part_owned_dof_ratio));
+    }
+
+    PetscCallMPI(MPI_Gather(&local_dofs, 1, MPIU_INT, gather_buffer, 1, MPIU_INT, 0, comm));
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
+      part_local_dofs[0]             = gather_buffer[0];              // min
+      part_local_dofs[1]             = gather_buffer[comm_size - 1];  // max
+      part_local_dofs[2]             = gather_buffer[median_index];   // median
+      PetscReal part_local_dof_ratio = (PetscReal)part_local_dofs[1] / (PetscReal)part_local_dofs[2];
+      PetscCall(PetscPrintf(
+          comm, "    Local %" PetscInt_FMT "-DoF nodes                  : %" PetscInt_FMT ", %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n", num_comp_q,
```

Comment:
Actually, not to keep bikeshedding, but maybe this would be better as "Partition nodes"? Feel like that might be more clear than "local".

Reply:
`LocalVector` has a precise meaning in PETSc and I think it's okay to use that. "Owned free" could be "Global" for consistency with `GlobalVector`.

Reply:
That would be fine with me.